aspnetcore/performance/caching/response.md

```diff
@@ -1,92 +1,72 @@
 
 ## ResponseCache Attribute
 
-The `ResponseCacheAttribute` is used to specify how a controller action's headers should be set to control its cache behavior. The attribute has the following properties, all of which are optional unless otherwise noted.
+The `ResponseCacheAttribute` specifies the parameters necessary for setting appropriate headers in response caching. See [ResponseCacheAttribute](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.aspnetcore.mvc.responsecacheattribute)  for a description of the parameters.
 
-**Duration `int`**
+`VaryByQueryKeys string[]` (requires ASP.NET Core 1.1.0 and higher): When set, the response caching middleware will vary the stored response by the given list of query keys. The middleware will serve the stored response only if the query keys of the request matches those of the original request that generated the stored response. The response caching middleware must be enabled to set the VaryByQueryKeys property, otherwise a runtime exception will be thrown.
 
-   The maximum duration (in seconds) the response should be cached. **Required** unless `NoStore` is `true`.
+There is no corresponding HTTP header for the `VaryByQueryKeys` property. This property is an HTTP feature handled by the response caching middleware. The response caching middleware is required to use the `VaryByQueryKeys` property.
 
-**Location `ResponseCacheLocation`**
+The `ResponseCacheAttribute` is used to configure and create (via `IFilterFactory`) a `ResponseCacheFilter`. The `ResponseCacheFilter` performs the work of updating the appropriate HTTP headers and features of the response. The filter:
 
-   The location where the response may be cached. May be `Any`, `None`, or `Client`. Default is `Any`.
+* Removes any existing headers for `Vary`, `Cache-Control`, and `Pragma`. 
+* Writes out the appropriate headers based on the properties set in the `ResponseCacheAttribute`. 
+* Updates the response caching HTTP feature if `VaryByQueryKeys` is set.
 
-**NoStore `bool`**
+### `Vary` 
 
-   Determines whether the value should be stored or not, and overrides other property values. When `true`, `Duration` is ignored and `Location` is ignored for values other than `None`.
+This header is only written when the `VaryByHeader` property is set. It is set to the `Vary` property's value. The following sample uses the `VaryByHeader` property.
 
-**VaryByHeader `string`**
+[!code-csharp[Main](response/sample/Controllers/HomeController.cs?name=snippet_VaryByHeader&highlight=1)]
 
-   When set, a `vary` response header will be written with the response.
+You can view the response headers with your browsers network tools or the [Fiddler tool](http://www.telerik.com/fiddler). It's instructive to compare the response headers on the first call to those on the second call when the request is cached. The following image shows the Edge F12 output on the **Network** tab when the `About2` action method is refreshed. 
```

Comment:
This section has a lot of nuances. For example, if you are using a browser to test this, I doubt you'll see any traffic on Fiddler since the browser will cache the response and no network traffic will be shown. Also, what are you expecting to be different between the first response and the second cached response? The headers should be identical in most cases.